src/language-html/parser-html.js

```diff
@@ -273,17 +278,19 @@ function _parse(text, options, parserOptions, shouldParseFrontMatter = true) {
     ? parseFrontMatter(text)
     : { frontMatter: null, content: text };
 
+  const file = new ParseSourceFile(text, options.filepath);
+  const start = new ParseLocation(file, 0, 0, 0);
+  const end = start.moveBy(text.length);
   const rawAst = {
     type: "root",
-    sourceSpan: { start: { offset: 0 }, end: { offset: text.length } },
+    sourceSpan: new ParseSourceSpan(start, end),
     children: ngHtmlParser(content, parserOptions, options),
   };
 
   if (frontMatter) {
-    frontMatter.sourceSpan = {
-      start: { offset: 0 },
-      end: { offset: frontMatter.raw.length },
-    };
+    const start = new ParseLocation(file, 0, 0, 0);
+    const end = start.moveBy(frontMatter.raw.length);
+    frontMatter.sourceSpan = new ParseSourceSpan(start, end);
```

Comment:
This is not related to linked issue, but improves the `sourceSpan` of "root" and "front-matter" Node